Prevent locked files assertion from taking a power assertion

bnham opened the pull request:
#### b77155f9e2ec8a25d434d9c35b9ce0b3df779484
<pre>
Prevent locked files assertion from taking a power assertion
<a href="https://bugs.webkit.org/show_bug.cgi?id=272009">https://bugs.webkit.org/show_bug.cgi?id=272009</a>
<a href="https://rdar.apple.com/125037124">rdar://125037124</a>

Reviewed by Per Arne Vollan and Chris Dumez.

The locked files assertion in NetworkProcess can sometimes take a power assertion, because
FinishTaskInterruptable assertions take power assertions. (There are cases where the system allows
you to create a FinishTaskInterruptable but doesn&apos;t actually immediately activate it, so it doesn&apos;t
*always* take a power assertion, but it can and does create a power assertion in various
circumstances.) This is bad because even taking and then immediately releasing a power assertion can
reset the idle timer for the whole device for several seconds.

To fix this, take a FinishTaskCanSleep assertion instead.

* Source/WebKit/NetworkProcess/NetworkProcess.cpp:
(WebKit::NetworkProcess::setIsHoldingLockedFiles):
* Source/WebKit/NetworkProcess/NetworkProcess.h:
* Source/WebKit/NetworkProcess/cocoa/NetworkProcessCocoa.mm:
(WebKit::NetworkProcess::acquireLockedFileActivity): Deleted.
(WebKit::NetworkProcess::invalidateFileActivity): Deleted.
(WebKit::NetworkProcess::hasAcquiredFileActivity const): Deleted.
* Source/WebKit/UIProcess/Cocoa/ProcessAssertionCocoa.mm:
(WebKit::runningBoardNameForAssertionType):
(WebKit::runningBoardDomainForAssertionType):
* Source/WebKit/UIProcess/ProcessAssertion.cpp:
(WebKit::processAssertionTypeDescription):
* Source/WebKit/UIProcess/ProcessAssertion.h:
* Source/WebKit/UIProcess/ProcessThrottler.cpp:
(WebKit::ProcessThrottler::assertionName const):

Canonical link: <a href="https://commits.webkit.org/276985@main">https://commits.webkit.org/276985@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b7f250c6b30d263b77dc5d1019d3d343b79ca85d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/46175 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/25306 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/48760 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/48846 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/42216 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/29662 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/22749 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/37730 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/46753 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/22397 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/39814 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/18936 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/19864 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/4219 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/42494 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/41261 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/50642 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/21171 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/17661 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/44904 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/22471 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/43812 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10255 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/22830 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/22165 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez reviewed (approved):
LGTM but please get a review from @pvollan too.

pvollan reviewed (approved):
Nice! R=me.

bnham committed:
Prevent locked files assertion from taking a power assertion
https://bugs.webkit.org/show_bug.cgi?id=272009
rdar://125037124

Reviewed by Per Arne Vollan and Chris Dumez.

The locked files assertion in NetworkProcess can sometimes take a power assertion, because
FinishTaskInterruptable assertions take power assertions. (There are cases where the system allows
you to create a FinishTaskInterruptable but doesn't actually immediately activate it, so it doesn't
*always* take a power assertion, but it can and does create a power assertion in various
circumstances.) This is bad because even taking and then immediately releasing a power assertion can
reset the idle timer for the whole device for several seconds.

To fix this, take a FinishTaskCanSleep assertion instead.

* Source/WebKit/NetworkProcess/NetworkProcess.cpp:
(WebKit::NetworkProcess::setIsHoldingLockedFiles):
* Source/WebKit/NetworkProcess/NetworkProcess.h:
* Source/WebKit/NetworkProcess/cocoa/NetworkProcessCocoa.mm:
(WebKit::NetworkProcess::acquireLockedFileActivity): Deleted.
(WebKit::NetworkProcess::invalidateFileActivity): Deleted.
(WebKit::NetworkProcess::hasAcquiredFileActivity const): Deleted.
* Source/WebKit/UIProcess/Cocoa/ProcessAssertionCocoa.mm:
(WebKit::runningBoardNameForAssertionType):
(WebKit::runningBoardDomainForAssertionType):
* Source/WebKit/UIProcess/ProcessAssertion.cpp:
(WebKit::processAssertionTypeDescription):
* Source/WebKit/UIProcess/ProcessAssertion.h:
* Source/WebKit/UIProcess/ProcessThrottler.cpp:
(WebKit::ProcessThrottler::assertionName const):

Canonical link: https://commits.webkit.org/276985@main
